internal/controller/maintenancemode_controller.go

```diff
@@ -77,50 +77,46 @@ func (r *MaintenanceModeReconciler) Reconcile(ctx context.Context, _ ctrl.Reques
 	r.log = log.FromContext(ctx)
 	r.log.Info("Starting reconcile")
 
-	nameToStorageClient := map[string]*v1alpha1.StorageClient{}
+	//TODO: This controller assumes that there is only one client and that the maintenance mode is created only for that
```

Comment:
will this be implemented in next version? if not, create a corresponding issue and tag to this PR for tracking.